docs/modules/maintain-cluster/pages/enterprise-rest-api.adoc

```diff
@@ -0,0 +1,411 @@
+
+[tabs] 
+==== 
+XML:: 
++ 
+-- 
+[source,xml]
+----
+<hazelcast>
+    <rest enabled="true">
+        <port>8080</port>
+    </rest>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  rest:
+    enabled: true
+    port: 8080
+----
+====
+
+== Authentication
+
+REST supports simple authentication. To configure authentication in the REST API, you should complete the following steps:
+
+* **#Enable Security:#**
+You need to enable the security to use REST API.
+
+* **#Configure Rest Realm:#**
+You need to introduce a REST realm as the authentication mechanism. This realm will handle user authentication using a simple method based on username and password.
+
+* **#Setup Users and Roles:#**
+You should identify users and their roles within the REST realm.
+
+* **#Configure AccessControlService:#**
+To configure `AccessControlService`, you should provide a factory class that creates an instance of the service. `AccessControlService` delegates authentication to the configured security realm. You can either create a custom factory class, or use the default implementation.
+For the default implementation, you can use `DefaultAccessControlServiceFactory`, to produce `DefaultAccessControlService`. Authorization decisions in this service are based on three predefined roles: `reader`, `writer` and `admin`.
+
+Following declarative example includes all the steps:
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    <security enabled="true">
+        <realms>
+            <realm name="restRealm">
+                <authentication>
+                    <simple>
+                        <user username="restuser" password="restsecret">
+                            <role>admin</role>
+                        </user>
+                    </simple>
+                </authentication>
+                <access-control-service>
+                    <factory-class-name>com.hazelcast.internal.rest.access.DefaultAccessControlServiceFactory</factory-class-name>
+                    <properties>
+                        <property name="decisionFile">/opt/acl.xml</property>
+                    </properties>
+                </access-control-service>
+            </realm>
+        </realms>
+    </security>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    realms:
+      - name: restRealm
+        authentication:
+          simple:
+            users:
+              - username: 'restuser'
+                password: 'restpassword'
+                roles:
+                  - admin
+        access-control-service:
+          factory-class-name: com.hazelcast.internal.rest.access.DefaultAccessControlServiceFactory
+          properties:
+            decisionFile: '/opt/acl.xml'
+----
+====
+
+== HTTPS Support
+You can configure SSL in the REST API. The configuration options are available to set up SSL as shown in
+the below declarative example:
```

Comment:
Can you add this info:

We use Spring Boot underneath to enable Spring web services, and all the SSL related options correspond to the underlying Spring Boot SSL support capabilities. 

The SSL configuration of the REST server is completely different from Hazelcast SSL configuration which has different configuration settings.

Reply:
fixed at https://github.com/hazelcast/hz-docs/pull/1097/commits/fc1ae49fe122c52714492f5263a82f3d5d930263